### DISCUSSION TITLE
WOLF FGB-K-28

### DISCUSSION THREAD

**Discussion by mzsolnay** [john30/ebusd#721] [Q&A]
### Description

Hi !

I'd like to use ebusd to communicate with a WOLF Heater, Type FGB-K-28.
Need help/assistance with configuration.

Thank you for taking the time helping me !

**Comment by tgrosz** [comment]
Hi, I see it was long time ago, but could you make it work?

**Reply by anxdroid** [reply]
I'm interested too, I'll try the ebusd v5 interface and let you know...

**Reply by tgrosz** [reply]
I made it work. I can query info but haven't tried writing to ebus

**Reply by anxdroid** [reply]
what kind of hardware are you using to query the bus ? I bought this https://adapter.ebusd.eu/v5/index.en.html and I'll use it with MQTT to make it available to home assistant

**Reply by tgrosz** [reply]
The same one, also with mqtt

**Reply by anxdroid** [reply]
I don't plan to write for the moment, I just need to have live data about the state, temperature, fails, amount of gas consumed

**Comment by anxdroid** [comment]
May I ask you to share the entities that home assistant or Just mqtt detects ? I am interested in tracking gas consumption and boiler status

**Comment by tgrosz** [comment]
[08.csv](https://github.com/user-attachments/files/17444397/08.csv)
[broadcast.csv](https://github.com/user-attachments/files/17444398/broadcast.csv)
[mqtt-hassio.cfg.txt](https://github.com/user-attachments/files/17444402/mqtt-hassio.cfg.txt)
I use these files. In mqtt config I changed only the filters (I guess) to include data from boiler.
Other 2 was translated (mostly) to English by the boiler's manual. I don't send the boiler config (HG values) because they are constant.
The boiler does not send gas consumption data as I know (I do not have any Wolf controllers, only the boiler). If you can get it, please share it.

**Reply by anxdroid** [reply]
Many thanks as soon as I can get It work I'll share all my findings

**Reply by tgrosz** [reply]
Could you find out more data? There are two I'm really interested in. First is the mode the boiler is in - idle is simple, but differentiate between hot water and heating. And I get 63 Celsius as outside temperature in broadcast - probably it is the not connected state. I'm thinking of connecting a digipot, but not sure it affects anything.

**Reply by anxdroid** [reply]
Hello I'm back :-) I finally got my new Wolf heater installed and I wanted to try the connection to mqtt/home assistant through the https://adapter.ebusd.eu/v5/index.en.html. So far I managed to connect the ebusd adapter to the heater (ebus + and - with no regard to polarity) using USB power and **no RPI**, hoping to get all I needed from MQTT. The adapter is connected to my home wifi network and send some info to my MQTT broker. 

Now I have two issues that I hope to get through if you can help me:

- on my MQTT broker I can only see a few info related to the adapter (temperature, uptime, RSSI, version, signal...) but nothing related to the heater
- in the adapter home page I have this alarming message:
Chip ID: ecda3b02342c, [ESP32-C3, rev 4](http://192.168.1.36/#/status?verbose)
Hostname: ebus-02342c
Up time: 181
Free heap: 87676 / 255652
**ebusd connected: no**
eBUS signal: acquired
MQTT connected: yes
- I hoped to see some data popping up in the eas MQTT topic just plugging the adapter to the heater but it is not happening. Should I provide some extended configuration file to the adapter? Should I send some command through the network or easy interface to start polling data ? Thank you so much !

**Reply by anxdroid** [reply]
some update: I finally managed to connect to the heater using ebusd addon in HASS. This is the log:

_in /config/ebusd-configuration/latest/en I put the files you linked (thank you !)_

> ebusd --foreground --httpport=8889 --mqtthost=192.168.1.5 --mqttport=1883 --mqttuser=_xxx_ --mqttpass=_xxx_ --scanconfig --mqttjson --configpath=/config/ebusd-configuration/latest/en --latency=9 --mqttint=/config/ebusd-configuration/mqtt-hassio.cfg --mqtttopic=ebusd --device=ens:192.168.1.36:9999 --log=all:notice --log=bus:notice --log=update:notice --log=network:notice
2024-11-09 16:58:15.227 [main notice] ebusd 23.2.p20230716 started with auto scan on device: 192.168.1.36:9999, enhanced
**2024-11-09 16:58:15.228 [main error] error reading config files from /config/ebusd-configuration/latest/en/: ERR: element not found, last error: 08.csv:3: ERR: element not found, field type TEMP in field 2**
2024-11-09 16:58:15.256 [bus notice] device status: resetting
2024-11-09 16:58:15.256 [bus notice] bus started with own address 31/36
2024-11-09 16:58:15.260 [bus notice] device status: reset, supports info
2024-11-09 16:58:15.260 [mqtt notice] connection established
2024-11-09 16:58:15.265 [bus notice] device status: extra info: firmware 1.1[4a1b].1[4a1b], jumpers 0x0b
2024-11-09 16:58:15.268 [bus notice] signal acquired
2024-11-09 16:58:18.325 [bus notice] new master 03, master count 2
2024-11-09 16:58:18.326 [update notice] received unknown BC cmd: 03fe050308010000005428ff3f
2024-11-09 16:58:25.496 [update notice] store 08 ident: ERR: argument value out of valid range
2024-11-09 16:58:25.497 [update error] unable to parse scan-read scan.08  from 3108070400 / 0a410126002000002a0001: ERR: argument value out of valid range
**2024-11-09 16:58:25.497 [bus notice] scan 08: ;65; &;**
2024-11-09 16:58:25.497 [main error] scan config 08: ERR: argument value out of valid range
2024-11-09 16:58:26.498 [mqtt error] decode scan.08 : ERR: argument value out of valid range
2024-11-09 16:58:28.431 [update notice] received unknown BC cmd: 03fe050308010000005428ff3f

still no useful info and as I'm reading this correctly, ebus cannot convert messages. Any idea ?

**Reply by anxdroid** [reply]
...and finally I did it !! There was the _template file missing in the local dir and all the data appeared:
<img width="352" alt="Screenshot 2024-11-09 alle 18 25 13" src="https://github.com/user-attachments/assets/fc069aec-2fb5-4954-ad53-c3cc36ba129f">
now, I'll try to extract something more and share any findings

**Reply by tgrosz** [reply]
I assembled an excel (will share soon) from a wolf pc app. It has a few more commands, but something is missing. Ebusd config has 3 bytes after the 5022 command but this source has only two - still that two matches the 2nd,3rd of ebusd bytes.

**Reply by tgrosz** [reply]
And one more thing: if you change the UIN to SIN where the value is 32768, then they'll undefined which is better I think

**Reply by andrelung** [reply]
Did you share the excel somewhere? I can't find it.
If I use your initial csvs, I get the following errors:
```
[main error] error reading common config file encon/broadcast.csv: ERR: element not found, encon/broadcast.csv:3: ERR: element not found, field type PERCENT0 in field 2
[main error] error reading scan config file encon/08.csv for ID "", SW65535, HW65535: ERR: element not found, encon/08.csv:3: ERR: element not found, field type TEMP in field 2
```

Any idea what I am doing wrong?

**Reply by tgrosz** [reply]
I found somewhere a better database
[wolf.json](https://github.com/user-attachments/files/24468581/wolf.json)
It has a section for FGB, unfortunately I couldn't make it work. I can check the values already in ebusd, but none of the others work. Probably I'm missing something.

**Reply by andrelung** [reply]
I think you got it from here: https://github.com/john30/ebusd/discussions/779#discussioncomment-14156249
The thing is, that that is no CSV file, it is just a database extracted from a WOLF app. Your heating actually seems to be included there:
250000 - FGB
Search for `"name": "FGB",` - you find it in line 54259
Below that you find the command, for example will line 54369 tell you, that "CC0C00" is the read command for Outside temperature (54356).

However, I am having difficulties finding a csv format which works for a lot of entities. Therefore if you have or had something working with more than 5 entities, I'd love to take a peak :)

**Reply by tgrosz** [reply]
I tried those commands with ebusd - and got no reply to any of them. That's the problem, a step is missing somewhere.
What works are the basic water temp in/out commands mostly plus decoding the simple broadcast.

**Reply by andrelung** [reply]
I copy-pasted a lot of these commands today and for me and my CGU-2K almost every number entity shows up:
<img width="326" height="558" alt="image" src="https://github.com/user-attachments/assets/fd600792-fc42-4acb-a173-5261c5603040" />

What I can't get to work is anything else, like
- time
- status
- modes

For this any hints or working examples would be awesome. 

If you want to take a look at my config, look here:
https://github.com/andrelung/ebusd-conf_wolf-cgu-2k-18_wolf-rm-2/

**Reply by tgrosz** [reply]
Would you elaborate on 'copy-pasted these commands'? What have you done exactly?
I have those entities working:
<img width="319" height="366" alt="image" src="https://github.com/user-attachments/assets/8f5c6475-31c0-4653-932d-32c490c32ac3" />
<img width="318" height="318" alt="image" src="https://github.com/user-attachments/assets/ebf21b26-59c8-473c-a4e0-cc4d7f4446f8" />
For ebus addresses: both the heater and ebusd have one master and one slave address - and I think we have to send queries/commands to the slave destination address.

**Reply by andrelung** [reply]
Like I said here: 
https://github.com/john30/ebusd/discussions/721#discussioncomment-15454560
1. I took the wolf.json
2. I searched for my device
3. I took each entry and created a CSV line based on this


Let me show you a working example from my heating system:
The json-lines
```
  {
    "DTID": "110000",
    "name": "CGU-2",
    "parameters": [

[...]

      {
        "type": "NumericParameterDescriptor",
        "PTID": "100008",
        "name": {
          "DEU": "Vorlaufsolltemperatur",
          "ESP": "Temperatura de consigna de impulsión",
          "GBR": "Set flow temperature"
        },
        "RemoveConditionId": "False",
        "ReadOnlyConditionId": "True",
        "InactiveConditionId": "False",
        "IsSnapshotTransmitEnabled": "false",
        "Decimals": "1",
        "ControlType": "NumericInput",
        "converter": {
          "type": "NumericConverter16Template",
          "TelegramNr": "2",
          "Type": "SS10"
        },
        "read_command": "CC0200"
      },

```
turned to
`r,hc,flow_temperature_desired,Vorlauf Sollwert,,35,5022,CC0200,value,,UIN,10,°C,temperature`

So I simply copy-pasted the read_command into an already working line (and obviously changed name etc). I am pretty sure this should work for your setup, too. I mean, not this specific line, more the general look-methology.

Regarding your screenshot:
Can you copy paste the lines for `broadcast status`? I am struggling with getting any value that is 
- a broadcast 
- a non-numeric value
Maybe that let's me figure out more entities:)

**Reply by tgrosz** [reply]
[broadcast.csv](https://github.com/user-attachments/files/24563254/broadcast.csv)
it's my broadcast file. As I remember, I did not change anything - just changed it to English. 
[mqtt-hassio.txt](https://github.com/user-attachments/files/24563296/mqtt-hassio.txt)
I had to change a few things in mqtt config (attached, renamed to txt) because it filtered out many messages.

**Reply by andrelung** [reply]
One other thing I did was commenting out the CSV definition:
`#type,circuit,level,name,comment,qq,zz,pbsb,id,*name,part,type,divisor/values,unit,comment`

Somehow the definition I saw in examples seems to differ from the "normal" or default CSV definition. But it was at some point. A big help is looking at the add-on logs after restarting it. If there's any issue in your CSV file, there will be a hint:
<img width="1257" height="460" alt="image" src="https://github.com/user-attachments/assets/abc916f3-b421-4baf-bfea-8d31446b40c8" />

**Reply by andrelung** [reply]
Actually your broadcast.csv is giving me an error, too:
<img width="1256" height="248" alt="Bildschirmfoto 2026-01-12 um 13 19 12" src="https://github.com/user-attachments/assets/f9bd8c1d-9ac2-4641-b39d-cf913333edba" />

Is there no error in your logs?

**Reply by tgrosz** [reply]
No, here's my 
[_templates.csv](https://github.com/user-attachments/files/24563728/_templates.csv)
It has the percent0 defined.
I was trying the commands from wolf.json with ebusctl. Most of them gives 'ERR: element not found' - a few gives back result (unfortunately they are already known) and there's a few where the json command is different from what we have in 08.csv and our csv is the good one always.

**Reply by andrelung** [reply]
Awesome! This gives me the whole broadcast:
```
{
     "status": {"value": 7},
     "condition": {"value": "79"},
     "boiler_output": {"value": 25},
     "forward_temp": {"value": 65.0},
     "backward_temp": {"value": 128},
     "dhw_temp": {"value": null},
     "outside_temp": {"value": -40}
}
```
Thank you!

Where do you see "ERR: element not found" - I haven't seen this either in ebusd, nor mqtt, nor home assistant. 

For me and my heating every numeric value works. You can easily identify them with:
```
        "converter": {
          "type": "NumericConverter16Template",
          "TelegramNr": "2",
          "Type": "SS10"
        },
```
in the CSV they identify as `SIN` or `UIN`

I think for the other values, we need a different decoding. Seeing definitions `BCD`, `D2C`, `BDA`, ... in your _templates.csv gives me a lot of ideas to try.

**Reply by tgrosz** [reply]
I use ebusctl.
1. ssh into ebusd addon
2. ebusctl read hg16
here hg16 is the field name from 08.csv. Or
3. ebusctl read -f -h 08502203b95501
here we're reading, destination is 08, read command is 5022, we'll send 03 bytes and b95501 is the field we'd like to read. This read does not care about decoding, it gives back the encoded result. Unfortunately, it gives back error for the 'new' codes in wolf.json

**Reply by tgrosz** [reply]
My theory - with AI help.
I do not have linkpro or bm module, just the heater. Now the heater does not send or accept many commands, just broadcast and a few more.
If linkpro is installed, then it tells the heater that it's there - so the heater will start communicating.
If you have linkpro (I don't know if the control module works without linkpro or not), then I think if you capture the data on ebus after a restart, then there will be a sequence probably from the linkpro to the heater that informs it that linkpro is there.
Maybe if we send the sequence, then after the heater would reply for more commands.

**Reply by Ces1254** [reply]
If you have no room controller and no Wolflink interface, the traffic on eBus is limited to the broadcast messages sent by the heater, very few. You have no other option than actively polling the parameters you are interested in using a proper configuration file. I imagine that your heater is on eBus at address 03/08. Check the work done by [zivillian](https://github.com/zivillian/ism7mqtt/blob/8265271df0c7fddf044f61151740b627771c934c/PROTOCOL.md) or the config files I have prepared [here](https://github.com/Ces1254/Wolf_ebusd_config). They are for a heat pump, but most of the parameters are the same.

**Reply by tgrosz** [reply]
Thanks, that's what I'm saying - if we could replay the sequence that is sent by Wolflink, that may allow us to communicate better with the heater. I'll try what of your codes I get reply from the heater.

**Comment by ningyichensha** [comment]
I run ebusd for an ochsner heatpump.
Maybe my solution using hex statements and python works for you as well.

I use hex statements to pass the commands to the ebus. This works without using message definitions.

For example setting the mode of hk1 = heating cicuit 1, submit this statement in the raspberry terminal: 
ebusctl hex 1006230603b200480100
That returns "done" and changes the mode for the hk1 to automatic (third last byte = 1)
standby = 1006230603b200480000
manual = 1006230603b200480700

To make ebusctl accept hex input, add --enablehex in file: /etc/default/ebusd.
In my case the line is: EBUSD_OPTS="--enablehex -d ens:/dev/ttyAMA0 --latency=50"

To find out which hex code is needed to change the hk1 setting, I used the terminal of the heatpump and changed the hk1 setting a couple of times.
Then I looked at the log file (/media/user/rootfs/var/log/ebusd.log). The time stamps helped to locate the hex commands.

From the log file you copy the hex command without the source address, which is in the first and second byte.
Now you try the statement: ebusctl hex "your hexcode"
That should return the desired answer "done". Than you check the setting at the heater.

Next step is the implementation in python:
My code to change the mode of the heatpump is this:

import subprocess
...
befehl = subprocess.run("ebusctl hex 1006230603b200480100", shell = True ,capture_output=True)
antwort = befehl.stdout
out = antwort.decode()
out = out.strip()
if out == "done":
.....

example read hk1 temperature:
in the log file I found: received unknown MS cmd: 011506210400820048 / 0a02800d02e80300000301
deleting the source address (first and second byte) -> 1506210400820048
From the response take:
low byte: 03 = 3 + high byte: 01 = 256  ->  259 /10 = 25,9 C

in Python:
import subprocess
...
temp_vorlauf = subprocess.run("ebusctl hex 1506210400820048", shell = True ,capture_output=True)
vorlauf = temp_vorlauf.stdout
out = vorlauf.decode()
out_low = out[18:20] #low byte
out_high = out[20:22] #high byte
value = ''.join([out_high, out_low])
vorlauf = int(value,16) /10 #convert hex string to integer an divide by 10 --> hk1 temp in float

**Comment by cr3pt** [comment]
Maybe someone can share full config files (cfg and csv) for FGB model?
I have few's but I didnt see all...
My Wolf config: FGB-28, KM-2, BM-2, RM-2,

**Reply by andrelung** [reply]
I am currently working on a RM-2 config:
https://github.com/andrelung/ebusd-conf_wolf-cgu-2k-18_wolf-rm-2/blob/main/encon/rm-2.csv

I think someone here had a working BM-2 config:
https://github.com/john30/ebusd/issues/460#issuecomment-1028407719

**Reply by lildadou** [reply]
I have a CGB-K-24 with a BM-1 heat controler but maybe helps?
https://github.com/lildadou/ebusd-wolf-cgb-bm1/tree/main/config/kromschroeder

**Reply by Ces1254** [reply]
If it may help, you can check [here](https://github.com/Ces1254/Wolf_ebusd_config). Note the broadcast message decoding at the end of listener.csv. The 0800 message has 4 temperature values coded in D2B (not SIN divided by 10 as for the usual temperature registers), which include the outdoor temperature, the DHW setpoint, the flow setpoint, and something else that is not available in my system (a FHA heat pump). Also note that to read big numbers that do not fit in 2 bytes (>65538) you need to issue a read command on two register addresses and decode the read back using ULG, not UIN (eg, try cc2a022b02 to get the operating hours stored in the two 16bit registers 022a and 022b).

**Comment by realResi** [comment]
For the CGG-2 with BM, PBSB=0800 contains boiler setpoint (D0–D1, UIN16 LE/256), outside temp (D2–D3), operating status (D5: 0x00=heating, 0x40=setback) and active setpoint (D7). Passive b-type matching doesn't work in ebusd v26 for this message type though.

More findings in my CGG-2 config repo:
https://github.com/realResi/ebusd-wolf-cgg2
